lib/src/signed_video_openssl.c

```diff
@@ -319,48 +320,58 @@ openssl_hash_data(void *handle, const uint8_t *data, size_t data_size, uint8_t *
 
 /* Initializes EVP_MD_CTX in |handle| with |hash_algo.type|. */
 svrc_t
-openssl_init_hash(void *handle)
+openssl_init_hash(void *handle, bool use_primary_ctx)
 {
   if (!handle) return SV_INVALID_PARAMETER;
   openssl_crypto_t *self = (openssl_crypto_t *)handle;
   int ret = 0;
-
-  if (self->ctx) {
+  EVP_MD_CTX **ctx = use_primary_ctx ? &self->primary_ctx : &self->secondary_ctx;
+  if (*ctx) {
     // Message digest type already set in context. Initialize the hashing function.
-    ret = EVP_DigestInit_ex(self->ctx, NULL, NULL);
+    ret = EVP_DigestInit_ex(*ctx, NULL, NULL);
   } else {
     if (!self->hash_algo.type) return SV_INVALID_PARAMETER;
     // Create a new context and set message digest type.
-    self->ctx = EVP_MD_CTX_new();
-    if (!self->ctx) return SV_EXTERNAL_ERROR;
+    *ctx = EVP_MD_CTX_new();
+    if (!(*ctx)) return SV_EXTERNAL_ERROR;
     // Set a message digest type and initialize the hashing function.
-    ret = EVP_DigestInit_ex(self->ctx, self->hash_algo.type, NULL);
+    ret = EVP_DigestInit_ex(*ctx, self->hash_algo.type, NULL);
   }
-
   return ret == 1 ? SV_OK : SV_EXTERNAL_ERROR;
 }
 
 /* Updates EVP_MD_CTX in |handle| with |data|. */
 svrc_t
-openssl_update_hash(void *handle, const uint8_t *data, size_t data_size)
+openssl_update_hash(void *handle, const uint8_t *data, size_t data_size, bool use_primary_ctx)
 {
   if (!data || data_size == 0 || !handle) return SV_INVALID_PARAMETER;
+
   openssl_crypto_t *self = (openssl_crypto_t *)handle;
+  EVP_MD_CTX **ctx = use_primary_ctx ? &self->primary_ctx : &self->secondary_ctx;
+
+  // Check if the context is valid.
+  if (!(*ctx)) return SV_EXTERNAL_ERROR;
+
   // Update the "ongoing" hash with new data.
-  if (!self->ctx) return SV_EXTERNAL_ERROR;
-  return EVP_DigestUpdate(self->ctx, data, data_size) == 1 ? SV_OK : SV_EXTERNAL_ERROR;
+  return EVP_DigestUpdate(*ctx, data, data_size) == 1 ? SV_OK : SV_EXTERNAL_ERROR;
 }
 
 /* Finalizes EVP_MD_CTX in |handle| and writes result to |hash|. */
```

Comment:
Update description from "Finalizes EVP_MD_CTX ..." to "Finalizes a EVP_MD_CTX ..."